Homework14(Photos)/Sources/ViewControllers/AlbumViewController.swift

```diff
@@ -0,0 +1,259 @@
+                    heightDimension: .fractionalWidth(0.1))
+                let layoutSectionHeader = NSCollectionLayoutBoundarySupplementaryItem(
+                    layoutSize: layoutSectionHeaderSize,
+                    elementKind: UICollectionView.elementKindSectionHeader,
+                    alignment: .top)
+                layoutSection.boundarySupplementaryItems = [layoutSectionHeader]
+
+                return layoutSection
+            default:
+                let itemSize = NSCollectionLayoutSize(
+                    widthDimension: .fractionalWidth(1),
+                    heightDimension: .fractionalHeight(1))
+                let layoutItem = NSCollectionLayoutItem(layoutSize: itemSize)
+
+                let groupSize = NSCollectionLayoutSize(
+                    widthDimension: .fractionalWidth(1),
+                    heightDimension: .estimated(46))
+                let layoutGroup = NSCollectionLayoutGroup.vertical(layoutSize: groupSize, subitems: [layoutItem])
+
+                let layoutSection = NSCollectionLayoutSection(group: layoutGroup)
+                layoutSection.contentInsets = NSDirectionalEdgeInsets(
+                    top: 10,
+                    leading: 15,
+                    bottom: 20,
+                    trailing: 15)
+
+                // Header
+                let layoutSectionHeaderSize = NSCollectionLayoutSize(
+                    widthDimension: .fractionalWidth(1),
+                    heightDimension: .fractionalWidth(0.1))
+                let layoutSectionHeader = NSCollectionLayoutBoundarySupplementaryItem(
+                    layoutSize: layoutSectionHeaderSize,
+                    elementKind: UICollectionView.elementKindSectionHeader,
+                    alignment: .top)
+                layoutSection.boundarySupplementaryItems = [layoutSectionHeader]
+
+                return layoutSection
+            }
+        }
+    }
+}
+
+extension AlbumViewController: UICollectionViewDataSource {
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        4
+    }
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        switch section {
+        case 0:
+            return Model.firstSectionData.count
+        case 1:
+            return Model.secondSectionData.count
+        case 2:
+            return Model.thirdSecrionData.count
+        case 3:
+            return Model.fourthSectionData.count
+        default:
+            return 0
+        }
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        switch indexPath.section {
+        case 0:
+            let item = collectionView.dequeueReusableCell(withReuseIdentifier: AlbumCell.identifier, for: indexPath) as? AlbumCell
+            item?.prepareData(data: Model.firstSectionData[indexPath.item])
+            return item ?? UICollectionViewCell()
```

Comment:
Проворачиваем тот же трюк + катим через guard 
```suggestion
        // Тоже самое
        let sectionType = Sections(rawValue: indexPath.section)
        switch sectionType {
        case .first:
            // Кастим через guard

            guard let item = collectionView.dequeueReusableCell(withReuseIdentifier: AlbumCell.identifier, for: indexPath) as? AlbumCell else {
                return UICollectionViewCell()
            }
            item.prepareData(data: Model.firstSectionData[indexPath.item])
            return item
```